### ISSUE TITLE
False positive not-context-manager

### ISSUE THREAD

**Issue by MartinBasti** [PyCQA/pylint#1503]
### Steps to reproduce
1. Have a following file
```
$ cat lock.py 
import threading

lock = threading.Lock()

with lock:
    print(dir(lock))
```
2. $ pylint-3 lock.py
3. shows `lock.py:5: [E1129(not-context-manager), ] Context manager 'lock' doesn't implement __enter__ and __exit__.)
` error
### Current behavior
Error is reported
```
lock.py:5: [E1129(not-context-manager), ] Context manager 'lock' doesn't implement __enter__ and __exit__.)
```
but `Lock()` object contains `__enter__` and `__exit__`:
```
['__class__', '__delattr__', '__dir__', '__doc__', '__enter__', '__eq__', '__exit__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', 'acquire', 'acquire_lock', 'locked', 'locked_lock', 'release', 'release_lock']

```

### Expected behavior
No "not-context-manager" error

### pylint --version output
```
$ pylint-3 --version
pylint-3 1.7.1, 
astroid 1.5.2
Python 3.6.1 (default, May  9 2017, 15:52:14) 
[GCC 7.1.1 20170503 (Red Hat 7.1.1-1)]
```

Older version doesn't report this false positive error:
```
pylint 1.6.4, 
astroid 1.4.5
```

**Comment by rogalski** [comment]
Thanks for submitting a bug.

It's already fixed on master, see change PyCQA/astroid@386fe2baabfc1ef2848211c3140521fa314d8748

**Comment by MartinBasti** [comment]
Thank you
